train.py

```diff
@@ -0,0 +1,223 @@
+    for batch_idx, batch in progress_bar:
+        real_data = batch[0].to(cfg.system.device)
+        batch_size = real_data.size(0)
+
+        # trick 6 from: https://github.com/soumith/ganhacks
+        real_labels = random_labels(batch_size, 0.8, 1, cfg.system.device)
+        fake_labels = random_labels(batch_size, 0.0, 0.1, cfg.system.device)
+
+        # train discriminator
+        discriminator.zero_grad()
+        label = real_labels
+        real_data = real_data.view(real_data.shape[0], 1, real_data.shape[1]).to(cfg.system.device)
+        disc_real_output = discriminator(real_data).view(-1)
+        discriminator_error_real = criterion(disc_real_output, label)
+        discriminator_error_real.backward()
+        D_x = disc_real_output.mean().item()  # Mean discriminator output for real data
+
+        # train generator
+        noise = torch.randn(batch_size, cfg.generator.noise_size, cfg.data.channels, device=cfg.system.device)
+        fake_data = generator(noise)
+        label = fake_labels
+        disc_fake_output = discriminator(fake_data.detach()).view(-1)
+
+        discriminator_error_fake = criterion(disc_fake_output, label)
+        discriminator_error_fake.backward()
+        D_G_z1 = disc_fake_output.mean().item()  # Discriminator's average output when evaluating the fake data
+        discriminator_error = discriminator_error_real + discriminator_error_fake
+        disc_optimizer.step()
+        generator.zero_grad()
+
+        label = real_labels
+        disc_output_after_update = discriminator(fake_data).view(-1)
+        generator_error = criterion(disc_output_after_update, label)
+        generator_error.backward()
+        D_G_z2 = disc_output_after_update.mean().item()  # Discriminator's output after updating the generator, fake data
+        gen_optimizer.step()
+
+        # log to wandb
+        if batch_idx % cfg.train.log_interval == 0:
+            generator_gradients = average_gradient(generator)
+            wandb.log(
+                {
+                    "generator_error": generator_error.item(),
+                    "discriminator_error": discriminator_error.item(),
+                    "D_x": D_x,
+                    "D_G_z1": D_G_z1,
+                    "D_G_z2": D_G_z2,
+                    "generator/generator_gradients": generator_gradients,
+                },
+                commit=False,
+            )
+            fig = visualize_training(generator, fixed_noise, epoch, batch_idx, cfg.logger.chart_path)
+            wandb.log({"fixed noise": wandb.Image(fig)}, commit=True)
+            # TODO: Might add local saving here as well
+            plt.close(fig)
+
+    checkpoint = {
+        "epoch": epoch,
+        "generator_state_dict": generator.state_dict(),
+        "discriminator_state_dict": discriminator.state_dict(),
+        "gen_optimizer_state_dict": gen_optimizer.state_dict(),
+        "disc_optimizer_state_dict": disc_optimizer.state_dict(),
+        "config": cfg,
```

Comment:
```suggestion
        "config": OmegaConf.to_object(cfg),
```

That way we'll save values like `run_name` in the resolved form, and not something like `${now:%Y_%m_%d_%H_%M}` :)